src/cpp/core/system/LinuxResources.cpp

```diff
@@ -223,19 +244,38 @@ class CGroupsMemoryProvider : public LinuxMemoryProvider
 
    Error getTotalMemory(long *pTotalKb, MemoryProvider *pProvider)
    {
-      Error error = getCgroupMemoryStat("memory.limit_in_bytes", pTotalKb); 
+      Error error;
+      if (!isV2_)
```

Comment:
It might read cleaner to have a `CGroupsV2MemoryProvider` instead of having this switching behavior (here and elsewhere); this would also give us UI confirmation of which mode we're in.

Reply:
There would be a fair amount of duplication in the implementations, but I can refactor it into separate providers if you think that's a more comprehensible structure.

Reply:
I think you could avoid duplication by deriving from CGroupsMemoryProvider instead of from LinuxMemoryProvider. Up to you, though, just a suggestion.

Reply:
I experimented with various designs here but ultimately couldn't come with anything that was simpler or more readable than a switch.

Reply:
Okay, fair enough!